subsys/bluetooth/host/hci_core.c

```diff
@@ -274,7 +274,15 @@ int bt_hci_cmd_send_sync(u16_t opcode, struct net_buf *buf,
 	BT_DBG("opcode 0x%04x status 0x%02x", opcode, cmd(buf)->status);
 
 	if (cmd(buf)->status) {
-		err = -EIO;
+		switch (cmd(buf)->status) {
+		case BT_HCI_ERR_CONN_LIMIT_EXCEEDED:
+			err = -EMLINK;
```

Comment:
How about -ECONNREFUSED or -ENOBUFS or -ENOMEM?

Reply:
I like the first one, since it makes it clear that this is about connections and not advertising as such (which may not otherwise be clear to the user since the API in question is bt_le_adv_start)